base/int.jl

```diff
@@ -207,7 +207,7 @@ function mod(x::T, y::T) where T<:Integer
     y == -1 && return T(0)   # avoid potential overflow in fld
     return x - fld(x, y) * y
 end
-mod(x::Signed, y::Unsigned) = rem(y + unsigned(rem(x, y)), y)
+mod(x::BitSigned, y::Unsigned) = rem(y + unsigned(rem(x, y)), y)
```

Comment:
Rather than remove the general `mod(x::Signed, y::Unsigned) ` method, wouldn't it be better to add a specific method for `mod(x::BigInt, y::Unsigned)`?

Reply:
I'm not sure what would be better. If special casing just `BigInt` is preferred to switching `Signed` to `BitSigned` we probably want to revert some changes in #23473 such as: https://github.com/JuliaLang/julia/pull/23473/files#diff-a59f124e9ddd1942d5324499c0cd3fceR601

Reply:
IMHO using `BitSigned` is better, as we don't know what is a good implementation for the general case. The only example of a `Signed` type in base which is not `BitSigned` is `BigInt` I think, and this general definition fails for it, which should be a good indication I believe to only define those functions for the specific types we know it works with.